graphql-server/pkg/resolvers/reaction.go

```diff
@@ -0,0 +1,67 @@
+package resolvers
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+
+import (
+	"context"
+	"fmt"
+
+	pkgContext "github.com/oursky/likedao/pkg/context"
+	servererrors "github.com/oursky/likedao/pkg/errors"
+	graphql1 "github.com/oursky/likedao/pkg/generated/graphql"
+	"github.com/oursky/likedao/pkg/logging"
+	"github.com/oursky/likedao/pkg/models"
+)
+
+func (r *mutationResolver) SetReaction(ctx context.Context, input models.SetReactionInput) (*models.Reaction, error) {
+	userAddress := pkgContext.GetAuthedUserAddress(ctx)
+	targetType := new(models.ReactionTargetType)
+	err := targetType.UnmarshalGQL(input.TargetID.EntityType)
+	if err != nil {
+		return nil, servererrors.QueryError.NewError(ctx, fmt.Sprintf("failed to unmarshal target type: %v", err))
+	}
+
+	res, err := pkgContext.GetMutatorsFromCtx(ctx).Reaction.SetReaction(input.TargetID.ID, *targetType, userAddress, input.Reaction)
+	if err != nil {
+		return nil, servererrors.MutationError.NewError(ctx, fmt.Sprintf("failed to set reaction: %v", err))
+	}
+	return res, nil
+}
+
+func (r *mutationResolver) UnsetReaction(ctx context.Context, input models.UnsetReactionInput) (*models.Reaction, error) {
+	logger := logging.GetLogger(ctx)
+	userAddress := pkgContext.GetAuthedUserAddress(ctx)
+	targetType := new(models.ReactionTargetType)
+	err := targetType.UnmarshalGQL(input.TargetID.EntityType)
+	if err != nil {
+		return nil, servererrors.QueryError.NewError(ctx, fmt.Sprintf("failed to unmarshal target type: %v", err))
+	}
+
+	res, err := pkgContext.GetMutatorsFromCtx(ctx).Reaction.UnsetReaction(input.TargetID.ID, *targetType, userAddress)
+	if err != nil {
+		logger.Error(servererrors.MutationError.NewError(ctx, fmt.Sprintf("failed to unset reaction: %v", err)))
+		return nil, nil
+	}
+
+	return res, nil
+}
+
+func (r *reactionResolver) Target(ctx context.Context, obj *models.Reaction) (models.Node, error) {
+	// TODO: Handle other target types
```

Comment:
should inspect node type & panic / return error here for unsupported types first?